api/turing/api/deployment_controller.go

```diff
@@ -188,14 +188,23 @@ func (c RouterDeploymentController) deployRouterVersion(
 		}
 	}
 
+	expSvc := c.BaseController.AppContext.ExperimentsService
 	if routerVersion.ExperimentEngine.Type != models.ExperimentEngineTypeNop {
-		// If passkey has been set, decrypt it
-		if routerVersion.ExperimentEngine.Config.Client.Passkey != "" {
-			experimentPasskey, err = c.CryptoService.Decrypt(routerVersion.ExperimentEngine.Config.Client.Passkey)
+		if expSvc.IsStandardExperimentManager(string(routerVersion.ExperimentEngine.Type)) {
+			// Convert the config to the standard type
+			expConfig, err := expSvc.GetStandardExperimentConfig(routerVersion.ExperimentEngine.Config)
 			if err != nil {
 				return "", c.updateRouterVersionStatusToFailed(err, routerVersion)
 			}
+			// If passkey has been set, decrypt it
+			if expConfig.Client.Passkey != "" {
+				experimentPasskey, err = c.CryptoService.Decrypt(expConfig.Client.Passkey)
+				if err != nil {
+					return "", c.updateRouterVersionStatusToFailed(err, routerVersion)
+				}
+			}
 		}
+
 		// Get the deployable Router Config for the experiment
 		experimentConfig, err = c.ExperimentsService.GetExperimentRunnerConfig(
 			string(routerVersion.ExperimentEngine.Type),
```

Comment:
and here

Reply:
Same: https://github.com/gojek/turing/pull/118#discussion_r749078096